src/main/java/mate/academy/internetshop/dao/impl/UserDaoImpl.java

```diff
@@ -40,4 +40,9 @@ public User update(User user) {
     public boolean delete(Long id) {
         return Storage.users.removeIf(u -> u.getId().equals(id));
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream().filter(u -> u.getLogin().equals(login)).findFirst();
```

Comment:
```suggestion
        return Storage.users.stream()
        .filter(u -> u.getLogin().equals(login))
        .findFirst();
```
не забуваємо за формат запису стрімів

Reply:
точно, спасибо.